### ISSUE TITLE
Could not create file - issue only if "intermediate" folder is on a mounted drive

### ISSUE THREAD

**Issue by Chingato** [nzbget/nzbget#720]
Hi,

I am getting errors when the intermediate folder is on a mounted share. As soon as I move the intermediate directory to a local path everything works fine even if all other directories are still on the mounted share. The permissions should be correct.

**Installed version:** 21.0
installed on a Raspberry 3b (Debian 10.5 based OSMC installation)

Extract from error log - full details see below:
`Sun Nov  1 10:28:57 2020	ERROR	Could not create file /mnt/mount_fritz/usenet_dl/inter/TEST_NZB{{)3G!6M_t_D_q__$}}.#1/9.out.tmp: (null)`

**Directories - Download fails:**
`MainDir=/mnt/mount_fritz/usenet_dl`
`DestDir=${MainDir}/downloads`
`InterDir=${MainDir}/inter`
`NzbDir=${MainDir}/nzb_import`
`QueueDir=${MainDir}/queue`
`TempDir=${MainDir}/tmp`
`WebDir=${AppDir}/webui`
`ScriptDir=${AppDir}/scripts`
`LockFile=${MainDir}/nzbget.lock`
`LogFile=${MainDir}/nzbget.log`
`ConfigTemplate=${AppDir}/webui/nzbget.conf.template`

**Directories - Download successful:**
same as above with the following change:
`InterDir=/opt/usenet_dl/inter`

**Directory permissions**
The mounted folders:
`osmc@osmc:/mnt/mount_fritz/usenet_dl$ ls -l`
`drwxrwx--- 2 osmc osmc     0 Nov  1 10:36 downloads`
`drwxrwx--- 2 osmc osmc     0 Nov  1 10:36 inter`
`drwxrwx--- 2 osmc osmc     0 Nov  1 10:28 nzb_import`
`-rwxrwx--- 1 osmc osmc     5 Nov  1 10:35 nzbget.lock`
`-rwxrwx--- 1 osmc osmc 22739 Nov  1 10:36 nzbget.log`
`drwxrwx--- 2 osmc osmc     0 Nov  1 10:36 queue`
`drwxrwx--- 2 osmc osmc     0 Nov  1 10:28 tmp`

local folder:
`osmc@osmc:/opt/usenet_dl$ ls -l`
`drwxr-xr-x 2 osmc osmc 4096 Nov  1 10:36 inter`

Mount happens via /etc/fstab
`//192.168.253.1/FRITZ.NAS/HDD4TB/mount/ /mnt/mount_fritz/ cifs x-systemd.automount,noauto,rw,iocharset=utf8,username=*******,password=******,uid=osmc,gid=osmc,file_mode=0770,dir_mode=0770,vers=3.0 0 0`

**Logfiles:**
Error if intermediate is on mounted volume:
[log_with_error.txt](https://github.com/nzbget/nzbget/files/5470484/log_with_error.txt)
Successful download if intermediate is on local path
[log_successful_download.txt](https://github.com/nzbget/nzbget/files/5470495/log_successful_download.txt)

**Comment by hugbug** [comment]
It fails to allocate output file of the required size.

What you could try:
- disable **DirectWrite** if it's active;
- if you use nzbget installation package from nzbget home page try the one from your OS distribution or vice versa;
- try compiling nzbget for (on) your device.

Anyway having **InterDir** on a network drive is bad for performance. Too much data is going there and back. It's always better to have InterDir locally.

**Comment by Chingato** [comment]
Hi,

I installed via "nzbget-latest-bin-linux.run"

After disabling DirectWrite it now works and I can download. But why does this happen? Is this a nzbget bug(/missing feature)  / not supported by the mounted share / incorrect mount options used?

I like to move the directory to a network share to avoid too much write access to the SD card to extend the time until it faisl due to bad sectors.

Thanks.

**Comment by hugbug** [comment]
>how can I disable DirectWrite ? I installed via "nzbget-latest-bin-linux.run"

In nzbget web interface go to settings tab. At the top right of the page there is a search box. Type "DirectWrite" there.

SDCard is indeed bad for frequent writing. Maybe use an external HDD or SDD; maybe you have an old small SSD which you don't use anymore.

Alternatively you could install nzbget on your FritzBox directly if you have terminal access there.

**Comment by Chingato** [comment]
you are too fast in answering ;-) I found it and changed it already. Can you please also comment on why this happens. See question above in my edited comment.

**Comment by hugbug** [comment]
> Is this a nzbget bug(/missing feature) / not supported by the mounted share / incorrect mount options used?

Unfortunately the error message is empty. Not printing a proper message is probably an nzbget bug.

First nzbget creates an output file. Then it tries to set the size of the file using a special Linux function. That function relies on a special file system feature called "sparse files". It is very much possible that this feature doesn't work via network.

If sparse files doesn't work then nzbget fills the file with nulls.

So basically the creating of output files should always work. Although in a case when sparse files are not supported the performance will degrade significantly.

**Comment by Chingato** [comment]
seems like sparse file is not working. Just tested the following command and it does not work on the mounted share but on local path.
`dd if=/dev/zero of=sparse_file bs=1 count=0 seek=512M`

> So basically the creating of output files should always work. Although in a case when sparse files are not supported the performance will degrade significantly.

As this is not the case it looks like as we have a bug here. Can this be fixed in a future version?

**Comment by hugbug** [comment]
Fixed.

If you want to try the new version send me a note to nzbget@gmail.com and I'll send you the new installer package.

However as I told before using shares for InterDir is bad for performance. Using shares which don't support sparse files is twice as bad.
